packages/@lwc/ssr-compiler/src/compile-template/index.ts

```diff
@@ -61,18 +65,62 @@ const bExportTemplate = esTemplate<
     }
 `;
 
-export default function compileTemplate(src: string, filename: string, options: TransformOptions) {
-    const { root, warnings } = parse(src);
+function templateUsesDirective(root: Root, target: ElementDirective['name']): boolean {
+    return root.children.some(function hasDirective(child) {
+        return (
+            ('directives' in child && child.directives.some((dir) => dir.name === target)) ||
+            ('children' in child && child.children.some(hasDirective))
+        );
+    });
+}
+
+export default function compileTemplate(
+    src: string,
+    filename: string,
+    // Technically this is @lwc/compiler's TransformOptions, but we can't use that without
+    // introducing a circular dependency. @lwc/template-compiler's options are close enough.
```

Comment:
We should copy-paste our own options or move the options object into `@lwc/shared`.